browser/brave_wallet/asset_discovery_manager_unittest.cc

```diff
@@ -184,8 +271,259 @@ class AssetDiscoveryManagerUnitTest : public testing::Test {
+    keyring_service_->RestoreWallet(kMnemonic1, kPasswordBrave, false,
+                                    base::DoNothing());
+
+    url_loader_factory_.SetInterceptor(base::BindLambdaForTesting(
+        [&, requests](const network::ResourceRequest& request) {
+          for (auto const& [url, address_response_map] : requests) {
+            if (request.url.spec().find("nfts") != std::string::npos) {
+              continue;
+            }
+            std::string header_value;
+            EXPECT_TRUE(
+                request.headers.GetHeader("X-Eth-Method", &header_value));
+
+            if (request.url.spec() == url.spec() &&
+                header_value == "eth_getLogs") {
+              base::StringPiece request_string(
+                  request.request_body->elements()
+                      ->at(0)
+                      .As<network::DataElementBytes>()
+                      .AsStringPiece());
+              std::string response;
+              for (auto const& [address, potential_response] :
+                   address_response_map) {
+                if (request_string.find(address.substr(2)) !=
+                    std::string::npos) {
+                  response = potential_response;
+                  break;
+                }
+              }
+              ASSERT_FALSE(response.empty());
+              url_loader_factory_.ClearResponses();
+              url_loader_factory_.AddResponse(request.url.spec(), response);
+            }
+          }
+        }));
+
+    std::size_t call_count{0};
+    while (eth_allowance_completed_call_count > call_count++) {
+      eth_allowance_manager_->DiscoverEthAllowancesOnAllSupportedChains();
+    }
+
+    wallet_service_observer_->WaitForOnDiscoverEthAllowancesCompleted(
+        contract_addresses, expected_allowances_count);
+  }
+
+  PrefService* GetPrefs() { return profile_->GetPrefs(); }
+  TestingPrefServiceSimple* GetLocalState() { return local_state_->Get(); }
+  GURL GetNetwork(const std::string& chain_id, mojom::CoinType coin) {
+    return brave_wallet::GetNetworkURL(GetPrefs(), chain_id, coin);
+  }
+
+  network::TestURLLoaderFactory url_loader_factory_;
+  scoped_refptr<network::SharedURLLoaderFactory> shared_url_loader_factory_;
+  std::unique_ptr<TestBraveWalletServiceObserverForAssetDiscoveryManager>
+      wallet_service_observer_;
+  content::BrowserTaskEnvironment task_environment_;
+  std::unique_ptr<ScopedTestingLocalState> local_state_;
+  std::unique_ptr<TestingProfile> profile_;
+  std::unique_ptr<BraveWalletService> wallet_service_;
+  std::unique_ptr<EthAllowanceManager> eth_allowance_manager_;
+  raw_ptr<KeyringService> keyring_service_ = nullptr;
+  JsonRpcService* json_rpc_service_;
+  TxService* tx_service;
+  base::test::ScopedFeatureList scoped_feature_list_;
+  data_decoder::test::InProcessDataDecoder in_process_data_decoder_;
 };
 
+TEST_F(EthAllowanceManagerUnitTest, AllowancesLoading) {
+  const std::string token_list_json = R"({
```

Comment:
This should be defined somewhere and shared instead of being copied in every test case.